onpremise/data_file_url.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"github.com/51Degrees/device-detection-go/v4/dd"
+	"github.com/51Degrees/device-detection-go/v4/onpremise"
+	"log"
+)
+
+func main() {
+	//Create config
+	config := dd.NewConfigHash(dd.Balanced)
+
+	//Create on-premise engine
+	pl, err := onpremise.New(
+		config,
+		onpremise.WithDataUpdateUrl(
+			"https://myprovider.com/1.tar.gz",
```

Comment:
Please see this example: https://github.com/51Degrees/device-detection-node/blob/main/fiftyone.devicedetection.onpremise/examples/onpremise/updatedatafile-console/updatedatafile.js:
1. let's name our updatedatafile.go - this is how it is named in other languages.  
2. let's make it accept a LicenseKey as environment variable,
3. if license key is not provided it should break with a console error message 
4. please add (or at least provide as comments) all possible options for the data file update mechanism and comment them - like in the JS example